commit-message-pr-style.md

```diff
@@ -0,0 +1,128 @@
+# Commit Message and Pull Request Style
+
+This document describes the commit message and Pull Request style applied to all PingCAP repositories. When you are to commit, be sure to follow the style to write a good commit message, a good Pull Request title and description.
+
+## Why a good commit message matters
+
+- To speed up the reviewing process
+    - Help the reviewers better understand the PR
+    - Allow ignoring unimportant commits
+- To help us write good Release Notes
+- To help the future maintainers establish the context
+    - Provide better information when browsing the history
+
+## What is a good commit message
+
+Elements of a good commit message:
+
+1. **What is your change? (mandatory)**
+
+    It can be fixing a specific bug, adding a feature, improving performance, reliability and stability, or just be a change for the sake of correctness.
+
+2. **Why this change was made? (mandatory)**
+
+    For short and obvious patches, this part can be omitted, but it should be a clear description of what the approach was.
+
+3. **What effect does the commit have? (optional)**
+
+    In addition to the obvious effects, this may include benchmarks, side effects, etc. For short and obvious patches, this part can be omitted.
+
+## How to write a good commit message
+
+To write a good commit message, we suggest following good format, cultivating good habits, and using good language.
+
+### Format of a good commit message
+
+Please follow the following style for **all your commits**:
+
+```
+<subsystem>: <what changed>
+<BLANK LINE>
+<why this change was made>
+<BLANK LINE>
+<footer>(optional)
+```
+
+- For the first subject line, use no more than 70 characters.
```

Comment:
Is this part duplicated with the content in CONTRIBUTING.md?

Reply:
@shenli Yes. This part is the same as that in CONTRIBUTING.md. This document is a more detailed version. Could we keep them both?